test/e2e/storage/drivers/csi.go

```diff
@@ -0,0 +1,243 @@
+}
+
+// gce-pd
+type gcePDCSIDriver struct {
+	controllerClusterRoles   []string
+	nodeClusterRoles         []string
+	controllerServiceAccount *v1.ServiceAccount
+	nodeServiceAccount       *v1.ServiceAccount
+
+	driverInfo DriverInfo
+}
+
+var _ TestDriver = &gcePDCSIDriver{}
+var _ DynamicPVTestDriver = &gcePDCSIDriver{}
+
+// InitGcePDCSIDriver returns gcePDCSIDriver that implements TestDriver interface
+func InitGcePDCSIDriver() TestDriver {
+	return &gcePDCSIDriver{
+		nodeClusterRoles: []string{
+			csiDriverRegistrarClusterRoleName,
+		},
+		controllerClusterRoles: []string{
+			csiExternalAttacherClusterRoleName,
+			csiExternalProvisionerClusterRoleName,
+		},
+		driverInfo: DriverInfo{
+			Name:        "csi-gce-pd",
+			FeatureTag:  "",
+			MaxFileSize: testpatterns.FileSizeMedium,
+			SupportedFsType: sets.NewString(
+				"", // Default fsType
+				"ext2",
+				"ext3",
+				"ext4",
+				"xfs",
+			),
+			IsPersistent:       true,
+			IsFsGroupSupported: true,
+			IsBlockSupported:   true,
+		},
+	}
+}
+
+func (g *gcePDCSIDriver) GetDriverInfo() *DriverInfo {
+	return &g.driverInfo
+}
+
+func (g *gcePDCSIDriver) SkipUnsupportedTest(pattern testpatterns.TestPattern) {
+	f := g.driverInfo.Framework
+	cs := f.ClientSet
+	config := g.driverInfo.Config
+	framework.SkipUnlessProviderIs("gce", "gke")
+	// TODO(#62561): Use credentials through external pod identity when that goes GA instead of downloading keys.
+	createGCESecrets(cs, config)
+	framework.SkipUnlessSecretExistsAfterWait(cs, "cloud-sa", config.Namespace, 3*time.Minute)
+}
+
+func (g *gcePDCSIDriver) GetDynamicProvisionStorageClass(fsType string) *storagev1.StorageClass {
+	f := g.driverInfo.Framework
+	cs := f.ClientSet
+	ns := g.driverInfo.Framework.Namespace.Name
+	provisioner := g.driverInfo.Name
+	suffix := fmt.Sprintf("%s-sc", g.driverInfo.Name)
+
+	// pods should be scheduled on the node
+	nodes := framework.GetReadySchedulableNodesOrDie(cs)
+	node := nodes.Items[rand.Intn(len(nodes.Items))]
+	g.driverInfo.Config.ClientNodeName = node.Name
+	g.driverInfo.Config.ServerNodeName = node.Name
+
+	nodeZone, ok := node.GetLabels()[kubeletapis.LabelZoneFailureDomain]
+	Expect(ok).To(BeTrue(), "Could not get label %v from node %v", kubeletapis.LabelZoneFailureDomain, node.GetName())
+	parameters := map[string]string{"type": "pd-standard", "zone": nodeZone}
+
+	return getStorageClass(provisioner, parameters, nil, ns, suffix)
+}
+
+func (g *gcePDCSIDriver) CreateDriver() {
+	By("deploying gce-pd driver")
+	f := g.driverInfo.Framework
+	cs := f.ClientSet
+	config := g.driverInfo.Config
+
+	g.controllerServiceAccount = csiServiceAccount(cs, config, "gce-controller", false /* setup */)
+	g.nodeServiceAccount = csiServiceAccount(cs, config, "gce-node", false /* setup */)
+	csiClusterRoleBindings(cs, config, false /* setup */, g.controllerServiceAccount, g.controllerClusterRoles)
+	csiClusterRoleBindings(cs, config, false /* setup */, g.nodeServiceAccount, g.nodeClusterRoles)
```

Comment:
we're missing the podsecuritypolicy bindings here, they are necessary as of 1.12+ as PSP is enabled by default  on CI